plugins/modules/aci_syslog_remote_dest.py

```diff
@@ -0,0 +1,340 @@
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+'''
+
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        name=dict(type='str', aliases=['remote_destination_name']),
+        format=dict(type='str', choices=['aci', 'nxos']),
+        admin_state=dict(type='str', choices=['enabled', 'disabled']),
+        description=dict(type='str'),
+        destination=dict(type='str'),
+        facility=dict(type='str', choices=['local0', 'local1', 'local2',
+                                           'local3', 'local4', 'local5',
+                                           'local6', 'local7']),
+        group=dict(type='str', aliases=['syslog_group', 'syslog_group_name']),
+        mgmt_epg=dict(type='str'),
+        syslog_port=dict(type='int'),
+        severity=dict(type='str', choices=['alerts', 'critical', 'debugging',
+                                           'emergencies', 'error',
+                                           'information', 'notifications',
+                                           'warnings']),
+        state=dict(type='str', default='present',
+                   choices=['absent', 'present', 'query'])
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['group', 'destination']],
+            ['state', 'present', ['group', 'destination']],
+        ]
+    )
+
+    aci = ACIModule(module)
+
+    name = module.params.get('name')
+    format = module.params.get('format')
+    admin_state = module.params.get('admin_state')
+    description = module.params.get('description')
+    destination = module.params.get('destination')
+    facility = module.params.get('facility')
+    group = module.params.get('group')
+    syslog_port = module.params.get('syslog_port')
+    severity = module.params.get('severity')
+    state = module.params.get('state')
+    mgmt_epg = module.params.get('mgmt_epg')
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class='syslogGroup',
+            aci_rn='fabric/slgroup-{0}'.format(group),
+            module_object=group,
+            target_filter={'name': group},
+        ),
+        subclass_1=dict(
+            aci_class='syslogRemoteDest',
+            aci_rn='rdst-{0}'.format(destination),
+            module_object=destination,
+            target_filter={'host': destination},
+        ),
+        child_classes=['fileRsARemoteHostToEpg'],
+    )
+
+    aci.get_existing()
+
+    if state == 'present':
+        if mgmt_epg:
```

Comment:
if mgmt_epg is None, I would prefer we don't build the child object at all. I would something like:
```
child_configs = []
if mgmt_epg:
    child_configs.append(dict(
        fileRsARemoteHostToEpg=dict(
            attributes=dict(
                tDn='uni/tn-mgmt/mgmtp-default/{0}'.format(mgmt_epg)
            ),
        ),
    ))
```

And then in the aci.payload I would do a child_configs = child_configs